server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -26,8 +28,18 @@
 public final class SearchRequestStats extends SearchRequestOperationsListener {
     Map<SearchPhaseName, StatsHolder> phaseStatsMap = new EnumMap<>(SearchPhaseName.class);
 
+    public static final String SEARCH_REQUEST_STATS_ENABLED_KEY = "search.request_stats_enabled";
+    public static final Setting<Boolean> SEARCH_REQUEST_STATS_ENABLED = Setting.boolSetting(
+        SEARCH_REQUEST_STATS_ENABLED_KEY,
+        false,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
     @Inject
-    public SearchRequestStats() {
+    public SearchRequestStats(ClusterSettings clusterSettings) {
```

Comment:
How are we ensuring requestStats is not called when disabled via cluster setting dynamically?
Considering we are now always initializing it at a node level. How are using this enabled variable to achieve that?

Reply:
We are using this SettingsUpdateConsumer to register the setEnable function to be called when the listener is disabled via cluster setting: https://github.com/opensearch-project/OpenSearch/pull/11526/files#diff-ec2e379635c68ce777a27bb04715c77c8358180e959c9028a8ec7bc738870a43R42

That will set the enable status to false for the global SearchRequestStats object. In that case, the `SearchRequestOperationsListeners` class won't return this listener when `buildCompositeListener` is called.